include/clang/AST/Decl.h

```diff
@@ -731,6 +734,25 @@ class DeclaratorDecl : public ValueDecl {
 
   friend class ASTDeclReader;
   friend class ASTDeclWriter;
+
+  // Checked C bounds information
+
+  /// \brief Return true if this declaration has bounds declared for it.
+  bool hasBoundsExpr() const;
+
+  /// \brief The declared bounds for this declaration. For function
+  /// declarations, this is the return bounds of the function. Null if no
+  /// bounds have been declared.
+  const BoundsExpr *getBoundsExpr() const {
```

Comment:
There are only 3 subclasses of DeclaratorDecl and it makes sense to set the bounds on all of them.